application/src/main/java/org/togetherjava/tjbot/commands/moderation/modmail/ModMailCommand.java

```diff
@@ -0,0 +1,160 @@
+import net.dv8tion.jda.api.entities.Guild;
+import net.dv8tion.jda.api.entities.MessageChannel;
+import net.dv8tion.jda.api.entities.MessageEmbed;
+import net.dv8tion.jda.api.entities.TextChannel;
+import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent;
+import net.dv8tion.jda.api.interactions.commands.OptionType;
+import net.dv8tion.jda.api.requests.restaction.MessageAction;
+import org.jetbrains.annotations.NotNull;
+import org.togetherjava.tjbot.commands.SlashCommandAdapter;
+import org.togetherjava.tjbot.commands.SlashCommandVisibility;
+import org.togetherjava.tjbot.config.Config;
+import org.togetherjava.tjbot.moderation.ModAuditLogWriter;
+
+import java.awt.*;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.List;
+import java.util.*;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Predicate;
+import java.util.regex.Pattern;
+
+/**
+ * Implements the '/modmail' command which can be used to automatically redirects messages from DM
+ * or Guild.
+ * <p>
+ * Actions are then logged to a dedicated channel, given by
+ * {@link Config#getModAuditLogChannelPattern()}.
+ */
+
+public class ModMailCommand extends SlashCommandAdapter {
+
+    private static final String COMMAND_NAME = "modmail";
+    private static final String MESSAGE = "message";
+    private static final String OPTION_MESSAGE_PRIVATE = "message-private";
+    private static final int COOLDOWN_DURATION_VALUE = 30;
+    private static final ChronoUnit COOLDOWN_DURATION_UNIT = ChronoUnit.MINUTES;
+    private final Cache<Long, Instant> channelIdToLastCommandInvocation;
+    private final Predicate<String> modMailChannelNamePredicate;
+
+
+    /**
+     * Creates a new instance.
+     *
+     * @param config the config to use for this
+     */
+    public ModMailCommand(@NotNull Config config) {
+        super(COMMAND_NAME, "Sends a message to the moderators", SlashCommandVisibility.GLOBAL);
+        getData().addOption(OptionType.STRING, MESSAGE, "Message to the moderators", true)
+            .addOption(OptionType.BOOLEAN, OPTION_MESSAGE_PRIVATE,
+                    "do you wish for your message to stay private", false);
+
+        channelIdToLastCommandInvocation = setCooldown();
+
+        modMailChannelNamePredicate =
+                Pattern.compile(config.getModMailChannelPattern()).asMatchPredicate();
+    }
+
+    @NotNull
+    private Cache<Long, Instant> setCooldown() {
+        return Caffeine.newBuilder()
+            .maximumSize(1_000)
+            .expireAfterAccess(COOLDOWN_DURATION_VALUE, TimeUnit.of(COOLDOWN_DURATION_UNIT))
+            .build();
+    }
+
+    @Override
+    public void onSlashCommand(@NotNull SlashCommandInteractionEvent event) {
+        MessageChannel messageChannel = event.getChannel();
+        if (isChannelOnCooldown(messageChannel)) {
+            event
+                .reply("Please wait a bit, this command can only be used once per %d %s.".formatted(
```

Comment:
Why not include how many second are left?

Reply:
Because I have just reused already existing method from Zabuzard to get the cooldown, but I can work on it right now.

Reply:
If you got some advice on how could I approach it, I would appreciate it : )

Reply:
Add a getCooldown method that returns an Integer (so null if non-existent)

And refactor the hasCooldown method (or whatever it's called) to use that + a null check?

Reply:
Like this?

```java
private boolean isChannelOnCooldown(@NotNull MessageChannel channel) {
        return Optional.ofNullable(channelIdToLastCommandInvocation.getIfPresent(channel.getIdLong())).map(
                        sinceCommandInvoked -> sinceCommandInvoked.plus(getCooldownDurationValue(),
                                                                        COOLDOWN_DURATION_UNIT))
                .filter(Instant.now()::isBefore).isPresent();
    }

    private Integer getCooldownDurationValue(){
        return COOLDOWN_DURATION_VALUE;
    }
```

Reply:
Done @Tais993